tests/ipsecondary_test.py

```diff
@@ -168,6 +170,42 @@ def test_primary_timeout_during_first_run(uptane_repo, secondary, aktualizr, **k
     return not_provisioned and not aktualizr.is_ecu_registered(secondary.id)
 
 
+@with_uptane_backend()
+@with_director()
+@with_secondary(start=False)
+@with_aktualizr(start=False, output_logs=True)
+def test_primary_wait_secondary_install(uptane_repo, secondary, aktualizr, director, **kwargs):
+    """Test that Primary waits for Secondary to connect before installing"""
+
+    # provision device with a secondary
+    with secondary, aktualizr:
+        aktualizr.wait_for_completion()
+
+    secondary_image_filename = "secondary_image_filename.img"
+    secondary_image_hash = uptane_repo.add_image(id=secondary.id, image_filename=secondary_image_filename)
+
+    end_ev = threading.Event()
+    def run_secondary():
+        time.sleep(10)
+        with secondary:
+            end_ev.wait()
+
+    sec_thread = threading.Thread(target=run_secondary)
+
+    with aktualizr:
+        sec_thread.start()
```

Comment:
why it cannot be 
`
with aktualizr:
   time.sleep(10)
   with secondary:
       aktualizr.wait_for_completion()

Reply:
yes, that's true. thanks!